tests/NCronJob.Tests/NotificationHandlerTests.cs

```diff
@@ -87,13 +73,24 @@ public async Task HandlerThatThrowsExceptionInAsyncPartShouldNotInfluenceOtherHa
                 .AddJob<ExceptionJob>(p => p.WithCronExpression(Cron.AtEveryMinute))
                 .AddNotificationHandler<HandlerThatThrowsInAsyncPartException>()
         );
-        var provider = CreateServiceProvider();
 
-        await provider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
+        await AssertSimpleJobWasProcessedAndNotified();
+    }
+
+    private async Task AssertSimpleJobWasProcessedAndNotified()
+    {
+        (IDisposable subscription, IList<ExecutionProgress> events) = RegisterAnExecutionProgressSubscriber(ServiceProvider);
+
+        await ServiceProvider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
```

Comment:
This part was very unexpected when I read the function name. Nothing indicates that it starts the `IHostedService`.

Reply:
Renamed to `StartNCronJobAndAssertSimpleJobWasProcessedAndNotified`